File: Training #2/Python/Diegogod/birthdates.py
from sys import stdin

def solve(names, days, months, years) -> tuple:

    youngest_index = max(range(len(names)), key=lambda i: (years[i], months[i], days[i]))
    oldest_index = min(range(len(names)), key=lambda i: (years[i], months[i], days[i]))

    return names[youngest_index], names[oldest_index]

def main():
    TC = int(stdin.readline().strip())  # Number of people in the class
    names, days, months, years = [], [], [], []

    for _ in range(TC):
        line = stdin.readline().strip().split()
        name = line[0]
        day, month, year = int(line[1]), int(line[2]), int(line[3])

        names.append(name)
        days.append(day)
        months.append(month)
        years.append(year)

    youngest, oldest = solve(names, days, months, years)

    # Print the youngest and oldest person
    print(youngest)
    print(oldest)

if __name__ == "__main__":
    main()

File: Trainings/Training #2/C++/Juanes/birthdates.cpp
#include <cstdio>
#include <tuple>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    vector<tuple<int, int, int, string>> people;

    for (int i = 0; i < n; i++) {
        char name[15];
        int day, month, year;
        scanf("%s %d %d %d", name, &day, &month, &year);

        people.push_back({year, month, day, string(name)});
    }

    auto youngest = max_element(people.begin(), people.end());
    auto oldest = min_element(people.begin(), people.end());
    
    printf("%s\n", get<3>(*youngest).c_str());
    printf("%s\n", get<3>(*oldest).c_str());

    return 0;
}
